SteamKit2/SteamKit2/Steam/Handlers/SteamAuthTicket/SteamAuthTicket.cs

```diff
@@ -55,13 +92,25 @@ public async Task<TicketInfo> GetAuthSessionTicket( uint appid )
 
             if ( GameConnectTokens.TryDequeue( out var token ) )
             {
-                var authTicket = BuildAuthTicket( token );
-                var ticket = await VerifyTicket( appid, authTicket, out var crc );
+                var authTicket = BuildAuthTicket( token, ticketType );
+                
+                // Steam add the 'str:' prefix to the identity string itself and appends a null terminator
+                var serverSecret = string.IsNullOrEmpty( identity ) ?  null : Encoding.UTF8.GetBytes( $"str:{identity}\0" );
+                var ticket = await VerifyTicket( appid, authTicket, serverSecret ,out var crc );
 
                 // Verify just in case
                 if ( ticket.ActiveTicketsCRC.Any( x => x == crc ) )
                 {
                     var tok = CombineTickets( authTicket, appTicket.Ticket );
+
+                    if (ticketType == TicketType.WebApiTicket)
```

Comment:
Should this be in CombineTickets to just directly create a correctly sized array?